rustfmt-core/rustfmt-config/src/lib.rs

```diff
@@ -556,36 +556,96 @@ ignore = []
-    //     ::std::env::set_var("CFG_RELEASE_CHANNEL", "nightly");
-    //     let mut config = Config::default();
-    //     config.set().unstable_features(true);
-    //     assert_eq!(config.was_set().unstable_features(), false);
-    //     config.set().unstable_features(true);
-    //     assert_eq!(config.unstable_features(), true);
-    //     ::std::env::set_var("CFG_RELEASE_CHANNEL", v);
-    // }
-
-    // #[test]
-    // fn test_unstable_from_toml() {
-    //     let mut config = Config::from_toml("unstable_features = true").unwrap();
-    //     assert_eq!(config.was_set().unstable_features(), false);
-    //     let v = ::std::env::var("CFG_RELEASE_CHANNEL").unwrap_or(String::from(""));
-    //     ::std::env::set_var("CFG_RELEASE_CHANNEL", "nightly");
-    //     config = Config::from_toml("unstable_features = true").unwrap();
-    //     assert_eq!(config.was_set().unstable_features(), true);
-    //     assert_eq!(config.unstable_features(), true);
-    //     ::std::env::set_var("CFG_RELEASE_CHANNEL", v);
-    // }
+    mod unstable_features {
+        use super::super::*;
+
+        #[test]
+        fn test_default_not_nightly_channel() {
+            if is_nightly_channel!() {
+                // This test requires non-nightly
+                return;
+            }
+            let config = Config::default();
+            assert_eq!(config.unstable_features(), false);
+            assert_eq!(config.was_set().unstable_features(), false);
+        }
+
+        #[test]
+        fn test_default_nightly_channel() {
+            if !is_nightly_channel!() {
+                // This test requires nightly
+                return;
+            }
+            let config = Config::default();
+            assert_eq!(config.unstable_features(), false);
+        }
+
+        #[test]
+        fn test_from_toml_not_nightly() {
+            if is_nightly_channel!() {
+                // This test requires non-nightly
+                return;
+            }
+            let config = Config::from_toml("unstable_features = true", Path::new("")).unwrap();
+            assert_eq!(config.was_set().unstable_features(), false);
+        }
+
+        #[test]
+        fn test_from_toml_nightly() {
+            if !is_nightly_channel!() {
+                // This test requires non-nightly
+                return;
+            }
+            let config = Config::from_toml("unstable_features = true", Path::new("")).unwrap();
+            assert_eq!(config.was_set().unstable_features(), true);
+        }
+
+        #[test]
+        fn test_set_not_nightly_channel() {
+            if is_nightly_channel!() {
+                // This test requires non-nightly
+                return;
+            }
+            let mut config = Config::default();
+            assert_eq!(config.unstable_features(), false);
+            config.set().unstable_features(true);
+            assert_eq!(config.unstable_features(), true);
```

Comment:
I didn't expect this to be the case, but apparently we don't check the stability attribute of config options for `set` and `override_value`. That seems to imply that folks can _technically_ already use unstable rustfmt options on stable anyway, though only via command line overrides to `rustfmt`

Reply:
Hmm, that's unfortunate. I will reconsider what we should do about unstable features on stable.